tests/step-103.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/sh
+# Test:
+#   Fail on not available shared hooks.
+
+if ! sh /var/lib/githooks/install.sh; then
+    echo "! Failed to execute the install script"
+    exit 1
+fi
+
+mkdir -p /tmp/shared/hooks-102.git/pre-commit &&
+    echo 'exit 0' >/tmp/shared/hooks-102.git/pre-commit/succeed &&
+    cd /tmp/shared/hooks-102.git &&
+    git init &&
+    git add . &&
+    git commit -m 'Initial commit' ||
+    exit 1
+
+# Install shared hook url into a repo.
+mkdir -p /tmp/test102 && cd /tmp/test102 || exit 1
+git init || exit 1
+mkdir -p .githooks && echo '/tmp/shared/hooks-102.git' >.githooks/.shared || exit 1
+git add .githooks/.shared
+git hooks shared update
+
+RESULT=$(ls -A ~/.githooks/shared 2>/dev/null)
+if [ "$RESULT" = "" ]; then
+    echo "! Expected shared hooks to be installed."
+    exit 1
+fi
+git commit -m "Test" || exit 1
+
+# Remove all shared hooks
+git hooks shared purge || exit 1
+
+# Clone a new one
+echo "Cloning"
+cd /tmp || exit 1
+git clone /tmp/test102 test102-clone && cd test102-clone || exit 1
+
+# Remove all shared hooks
+git hooks shared purge || exit 1
+
+echo "Commiting"
+# Make a commit
+echo A >A || exit 1
+git add A || exit 1
+OUTPUT=$(git commit -a -m "Test" 2>&1)
+
+# shellcheck disable=SC2181
+if [ $? -eq 0 ] || ! echo "$OUTPUT" | grep -q "Failed to execute shared hook"; then
+    echo "! Expected to fail on not availabe shared hooks. output:"
+    echo "$OUTPUT"
+    exit 1
+fi
+
+git hooks shared pull || exit 1
+
+# Change url
+(cd ~/.githooks/shared/shared_hooks_102* &&
+    git remote rm origin &&
+    git remote add origin /some/other/url.git) || exit 1
+# Make a commit
+echo A >>A || exit 1
+OUTPUT=$(git commit -a -m "Test" 2>&1)
+
+# shellcheck disable=SC2181
+if [ $? -eq 0 ] || ! (echo "$OUTPUT" | grep "Shared root url" | grep -q "not the same"); then
```

Comment:
Same as above: I expected this commit to work, and it did.

Reply:
Ok. Maybe we should make a fail flag in the settings, that any `fail-on-non-existing-shared-hook`
Thats the behavior I would enforce always :-) But maybe users dont want that??...

Reply:
I'd err on the side of being able to complete the task you want (you want to commit & push a change) even if there's a hiccup with one of the hooks. CI should catch those problems too, right? :)

Reply:
Ok. Continuing is the default behavior. But a flag would help in my setup.
- githooks.failOnNonExistingHooks = false
CI can not check everything, for example `default lfs hooks` which are reference in `.shared`
If you dont run these, you have a real problem. So I would really set `githooks.failOnNonExistingHooks = true`

> So if we can tell the git clone post-checkout events apart from the git checkout triggered ones, we could refresh shared hooks there as well.
Jeah, that would be good.
So we check for a null reference if its the `post-checkout` and only then update. Thats sounds good to me.